### DISCUSSION TITLE
LastFM API integration

### DISCUSSION THREAD

**Discussion by Scaffale** [blackcandy-org/blackcandy#224] [Ideas]
LastFM offers some interesting API such as:
* artist -> [top tracks](https://www.last.fm/api/show/artist.getTopTracks)
* artist -> [similar artists](https://www.last.fm/api/show/artist.getSimilar)
* song -> [similar songs](https://www.last.fm/api/show/track.getSimilar)

These are useful to easily and quickly create playlists.

It need and api_key that could be added in the settings. 
Also the partial loading such informations will probably fit better in an async render.

I'll start implementing these features, but I could use some help for code-formatting and frontend features...

Thanks!

**Comment by aidewoode** [comment]
Thanks for your interest to contribute black candy. But I not quite sure what features you actually want to make.
You said LastFM API is useful to easily and quickly create playlists. But how these API can help to create playlists easily? If you can explain more details about the features you want to make, it will be helpful.

**Reply by Scaffale** [reply]
Probably "playlist" isn't the proper term.
The idea is: you have the discography of an artist, you want to listen to some top hits without searching around on each album you have. Either you build up a playlist of your own, or you could use the "artists top song" api from lastfm.

The second use case could be: I like this artist, let me listen to similar artists in my library.

The last use case is similar to the second: I like this song, let me listen to similar songs.

Does this make sense?

As soon as I finish adding some tests I'll make a pull request so you can check the "top songs" logic, if you like, for I've implemented it.
